Exercise_nonsequential/array_job_nonsequence.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# This script runs the array job spawned using the main.sh script on a
+# single node on express partition to process folders starting with
+# name sub-blast located inside a work directory.
+
+#SBATCH -J slurm_python
+#SBATCH --partition=reservation                                                                                                                                                                                    
+#SBATCH --reservation=fall_training_cpu_2022
+#SBATCH -o %A_%a.o 
+#SBATCH -e %A_%a.e 
+#SBATCH --nodes=1
+#SBATCH --ntasks-per-node=1
+#SBATCH --cpus-per-task=1
+#SBATCH --time=00:20:00
+#SBATCH --mem=4G 
+
+# Use a regular partition, such as express, when not in a training                                                                                                                                                                  
+# session                                                                                                                                                                                                                   
+##SBATCH --partition=express
+
+module load anaconda3/2022.01
+
+# Change 'BASEDIR' and 'WORKDIR' based on desired locations
+BASEDIR=$PWD
+WORKDIR=$PWD/masks
+
+ls $WORKDIR | grep sub > filelist.txt
```

Comment:
Implicit in this is that the working directory contains sub-directories with `sub` in their names, but `filelist.txt` will include any file or directory with `sub` in its name.  This is of course just an example that already contains directories named `sub-*` but this pattern could lead to confusion and or unexpected behavior if employed for real workloads.

Reply:
@dhml , you are correct that this example is setup with a specific sub-directory structure in mind. The main point here was to show how users can use arrays for non-sequential data. So create a file list of entries based on a desired filter and then use that file list to execute the array command. 
If you think it would be prudent to add that explanation to the script, let me know and I can put those as comments in the array_job_nonsequence.sh script.

Reply:
@dhml , also, I have now added the 'find' command in main.sh & array_job_nonsequence.sh along with relevant comments after testing at my end. Please see the revised version.